sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/unity/identity/AzureSpringConfiguration.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.unity.identity;
+
+import com.azure.core.util.Configuration;
+import com.azure.spring.autoconfigure.unity.AzureProperties;
+import com.azure.spring.identity.CredentialPropertiesProvider;
+import org.springframework.util.StringUtils;
+
+/**
+ * Extends the {@link Configuration} to provide Azure Spring related configurations.
+ */
+public class AzureSpringConfiguration extends Configuration implements CredentialPropertiesProvider {
+
+    private final Configuration configuration;
+
+    public AzureSpringConfiguration(AzureProperties azureProperties) {
+        this.configuration = Configuration.getGlobalConfiguration().clone();
+        overridePropertyIfApplicable(Configuration.PROPERTY_AZURE_TENANT_ID, azureProperties.getCredential().getTenantId());
```

Comment:
check this util class `org.springframework.boot.context.properties.PropertyMapper`